Sources/StreamChatUI/ChatMessageList/Attachments/Poll/PollCreationVC/PollCreationFeatureCell.swift

```diff
@@ -0,0 +1,75 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+/// The cell for enabling or disabling a poll feature.
+open class PollCreationFeatureCell: _CollectionViewCell, ThemeProvider {
+    public struct Content {
+        public var feature: PollFeature
+
+        public init(feature: PollFeature) {
+            self.feature = feature
+        }
+    }
+
+    public var content: Content? {
+        didSet {
+            updateContentIfNeeded()
+        }
+    }
+
+    /// The main container that holds the subviews.
+    open private(set) lazy var container = HContainer()
+
+    /// A view that displays the feature name and the switch to enable/disable the feature.
+    open private(set) lazy var featureSwitchView = components
+        .pollCreationFeatureSwitchView.init()
+        .withoutAutoresizingMaskConstraints
+
+    /// A closure that is triggered whenever the switch value changes.
+    public var onValueChange: ((Bool) -> Void)?
+
+    override open func setUp() {
+        super.setUp()
+
+        contentView.isUserInteractionEnabled = true
+        featureSwitchView.onValueChange = { [weak self] newValue in
+            self?.onValueChange?(newValue)
+        }
+    }
+
+    override open func setUpAppearance() {
+        super.setUpAppearance()
+
+        backgroundColor = appearance.colorPalette.background
+        container.backgroundColor = appearance.colorPalette.background1
+        container.layer.cornerRadius = 16
+    }
+
+    override open func setUpLayout() {
+        super.setUpLayout()
+
+        container.views {
+            featureSwitchView
+        }
+        .height(56)
```

Comment:
Since the height is constant 56, it should not change. Can you share a video of that happening?

Reply:
Ah you mean the value is present in multiple views? 😂

Reply:
TBH I'm not sure there is much value in extracting these values, because the views are quite simple, and it happens that in our design, the textfields/cells have the same height, but this could totally change in the customers app. Since most likely they will override the whole layout function, they will set their own constant. 

Either way, I've done this already, let me know what you think of the implementation.

Reply:
looks good